src/pages/blog/streamlit-deployment-https-social-auth.md

```diff
@@ -0,0 +1,122 @@
+---
+title: 'Deploy Streamlit Applications with Gimlet'
+date: '2024-07-10'
+description: Find out how to deploy Streamlit apps with HTTPS certificate and social authentication using Gimlet.
+image: streamlit-deploy-gimlet-https-social-authentication.jpg
+toc: false
+---
+
+**Streamlit Community Cloud is on par with the simplicity of the popular framework itself. It’s awesome to have a hosting solution dedicated to it. This comes with some limitations, however, which we aim to overcome with Gimlet’s upcoming 1.0 version.**
+
+Streamlit is an amazing framework for data scientists and people who work with language models. It has little to no learning curve, just like Streamlit Community Cloud.
+
+Gimlet 1.0 release is around the corner. It comes with several hosting related capabilities that’s useful for data scientists working with Streamlit applications every day.
+
+## Deployment of Streamlit Apps with Gimlet
+
+Deploying a Streamlit application can be done in X number of steps.
+
+You can get started with Gimlet by connecting your GitHub account. After that, add the repository that has your Streamlit application by clicking the **Import** button next to it. Save the added repo by clicking the **I am done importing** button. If you can’t find it on the top of the list, use the search bar.
+
+Navigate to the deployment setup screen by clicking on the card of your Streamlit application. Set up the deployment in the following way:
+
+- **Deployment template:** Web Application. This’ll allow you to select the Dockerfile container image setting.
+- **Container Image:** Using a Dockerfile. This method requires a Dockerfile. If you don’t have one, the example below can be tailored to your app or used without changes. If you’re not familiar with Dockerfiles, read the comments.
+
+```
+# app/Dockerfile
+
+# This Dockerfile uses python:3.9-slim, which is a lightweight implementation of the Python image. This’ll reduce image size.
+FROM python:3.9-slim
+
+# Sets the working directory inside the container to `/app`. It’s unlikely that you’ll need to adjust this.
+WORKDIR /app
+
+# Installs several packages and cleans up the apt cache. In some cases some of these packages might not be needed to run a Streamlit container.
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    curl \
+    software-properties-common \
+    git \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copies everything from the root directory to the working directory of the container. This is usually used for development environments.
+COPY . .
```

Comment:
Let's revise this together. This will also copy .env files and everything into the container.